base/irrationals.jl

```diff
@@ -221,6 +221,16 @@ end
 log(::Irrational{:e}) = 1 # use 1 to correctly promote expressions like log(x)/log(e)
 log(::Irrational{:e}, x::Number) = log(x)
 
+# These methods should go to special/trig.jl, but Irrational is not yet defined when the
+# file is loaded.
+sinpi(x::Irrational) = sinpi(float(x))
+cospi(x::Irrational) = cospi(float(x))
+sinc(x::Irrational) = sinc(float(x))
+sinc(z::Complex{T}) where {T<:Irrational} = sinc(float(z))
```

Comment:
what sacha suggest is just sugar for what you have. if it is dispatching to the wrong method then it is a bug somewhere